e2e/e2e_test.go

```diff
@@ -451,3 +452,50 @@ spec:
 		})
 	}
 }
+
+func getPipelineDataPath(subdir string) string {
+	testdatapath, _ := testutil.GetTestDataPath()
+	return filepath.Join(
+		testdatapath,
+		"pipeline",
+		subdir,
+	)
+}
+
+func TestKptPipeline(t *testing.T) {
+	tests := map[string]struct {
+		// may need more fields in the future
+		dataset      string
+		command      string
+		args         []string
+	}{
+		"basic pipeline run command": {
+			dataset: testutil.BasicPipeline,
+			command: "run",
+		},
+	}
+
+	for _, test := range tests {
+		dir, err := ioutil.TempDir("", "")
+		if !assert.NoError(t, err) {
+			t.FailNow()
+		}
+
+		testdata := filepath.Join(getPipelineDataPath("testdata"), test.dataset)
+		golden := filepath.Join(getPipelineDataPath("golden"), test.dataset)
```

Comment:
Is using a golden directory structure for the output scalable here? Our experience with packaging is that there are many cases to test and that providing a large number of golden results is hard to maintain.